File: sdk/samples/all in on code/Visual Studio 2008/VBWinFormDesigner/Adornment/UC_AdornmentPainting.vb
'************************************* Module Header **************************************\
' Module Name:	UC_AdornmentPainting.vb
' Project:		VBWinFormDesigner
' Copyright (c) Microsoft Corporation.
' 
' 
' The AdornmentPainting sample demonstrates how to paint additional adornments on top of 
' the control.  
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.

' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************'

Imports System.Windows.Forms.Design
Imports System.Drawing.Drawing2D
Imports System.ComponentModel

Namespace Adornment
    <Designer(GetType(UC_AdornmentPaintingDesigner))> _
    Public Class UC_AdornmentPainting

    End Class

    Friend Class UC_AdornmentPaintingDesigner
        Inherits ControlDesigner

        Protected Overrides Sub OnPaintAdornments(ByVal pe As PaintEventArgs)
            Dim uc As UC_AdornmentPainting = DirectCast(MyBase.Component, UC_AdornmentPainting)
            Dim clientRectangle As Rectangle = uc.ClientRectangle
            Using pen As Pen = New Pen(Color.Red)
                pen.DashStyle = DashStyle.Dot
                clientRectangle.Width -= 1
                clientRectangle.Height -= 1
                pe.Graphics.DrawRectangle(pen, clientRectangle)
            End Using
            MyBase.OnPaintAdornments(pe)
        End Sub

    End Class
End Namespace

File: sdk/samples/all in on code/Visual Studio 2008/CSWinFormDesigner/Adornment/UC_AdornmentPainting.cs
/************************************* Module Header **************************************\
* Module Name:	UC_AdornmentPainting.cs
* Project:		CSWinFormDesigner
* Copyright (c) Microsoft Corporation.
* 
* 
* The AdornmentPainting sample demonstrates how to paint additional adornments on top of 
* the control.  
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* History:
* * 4/25/2009 3:00 PM Zhi-Xin Ye Created
* * 
* 
\******************************************************************************************/


using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Windows.Forms.Design;
using System.Drawing.Drawing2D;

namespace CSWinFormDesigner.Adornment
{
    [Designer(typeof(UC_AdornmentPaintingDesigner))]
    public partial class UC_AdornmentPainting : UserControl
    {
        public UC_AdornmentPainting()
        {
            InitializeComponent();
        }
    }

    class UC_AdornmentPaintingDesigner : ControlDesigner
    {
        protected override void OnPaintAdornments(PaintEventArgs pe)
        {
            UC_AdornmentPainting uc = (UC_AdornmentPainting)Component;
            
            Rectangle clientRectangle = uc.ClientRectangle;

            //TODO: Code here to paint adornments on the top the control 

            using (Pen pen = new Pen(Color.Red))
            {
                pen.DashStyle = DashStyle.Dot;
                clientRectangle.Width -= 1;
                clientRectangle.Height -= 1;
                pe.Graphics.DrawRectangle(pen, clientRectangle);
            }

            base.OnPaintAdornments(pe);
        }

    }
}
